models/gif_psc_exp.cpp

```diff
@@ -294,26 +290,21 @@ nest::gif_psc_exp::calibrate()
   // since t_ref_ >= 0, this can only fail in error
   assert( V_.RefractoryCounts_ >= 0 );
 
-  // initializing internal state
-  if ( !S_.sfa_stc_initialized_ )
-  {
-    V_.P_sfa_.clear();
-    V_.P_stc_.clear();
-
-    for ( size_t i = 0; i < P_.tau_sfa_.size(); i++ )
-    {
-      V_.P_sfa_.push_back( std::exp( -h / P_.tau_sfa_[ i ] ) );
-    }
-    S_.sfa_elems_.resize( P_.tau_sfa_.size(), 0.0 );
+  // initializing adaptation (stc/sfa) variables
+  V_.P_sfa_.clear();
+  V_.P_stc_.clear();
 
-    for ( size_t i = 0; i < P_.tau_stc_.size(); i++ )
-    {
-      V_.P_stc_.push_back( std::exp( -h / P_.tau_stc_[ i ] ) );
-    }
-    S_.stc_elems_.resize( P_.tau_stc_.size(), 0.0 );
+  for ( size_t i = 0; i < P_.tau_sfa_.size(); i++ )
+  {
+    V_.P_sfa_.push_back( std::exp( -h / P_.tau_sfa_[ i ] ) );
+  }
+  S_.sfa_elems_.resize( P_.tau_sfa_.size(), 0.0 );
```

Comment:
This is safe: New elements with value 0 are only added if more time constants have been added, and if the number of time constants is reduced, extra elements are discarded. You should add a comment to `init_state_()` that `sfa_elems_` is initialized here. 

I still wonder a bit about the use case: adding or removing components from the function describing the SFA time course during a break in the simulation. An alternative would be to forbid changing then number of time constants once the simulation has started (in `State::set_status()`).